File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/definitive/DarknessModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.definitive;

import dagger.Binds;
import dagger.Module;
import dagger.Reusable;
import se.gustavkarlsson.skylight.android.background.providers.DarknessProvider;
import se.gustavkarlsson.skylight.android.background.providers.impl.KlausBrunnerDarknessProvider;

@Module
public abstract class DarknessModule {

	@Binds
	@Reusable
	abstract DarknessProvider bindDarknessProvider(KlausBrunnerDarknessProvider provider);

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/definitive/DarknessModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.definitive

import dagger.Binds
import dagger.Module
import dagger.Reusable
import se.gustavkarlsson.skylight.android.services.providers.DarknessProvider
import se.gustavkarlsson.skylight.android.services_impl.providers.KlausBrunnerDarknessProvider

@Module
abstract class DarknessModule {

    @Binds
    @Reusable
    abstract fun bindDarknessProvider(impl: KlausBrunnerDarknessProvider): DarknessProvider

}
